### DISCUSSION TITLE
Does egui_plot trigger a repaint ? Can this be avoided ?

### DISCUSSION THREAD

**Discussion by vortex314** [emilk/egui#3991] [Q&A]
When adding this code to my dashboard, it looks to trigger a repaint which triggers a redraw which triggers a repaint and so on ......
Is this expected behavior or am I missing something in the logic ? 
Just to avoid that my application consumes cpu for no update what so ever. 
```rust
    fn draw(&mut self, ui: &mut Ui) -> Result<(), String> {
        let s = format!("{} {}", self.value, self.unit);
        let rect = rect_border(self.rect);
        let n = 128;
        let line_points: PlotPoints = (0..=n)
            .map(|i| {
                use std::f64::consts::TAU;
                let x = egui::remap(i as f64, 0.0..=n as f64, -TAU..=TAU);
                [x, x.sin()]
            })
            .collect();
        let line = egui_plot::Line::new(line_points);
        let pl = egui_plot::Plot::new("example_plot")
            .height(rect.height())
            .width(rect.width())
            .show_axes(true)
            .show_grid(true)
            .data_aspect(1.0);
        let layout = Layout::top_down(Align::LEFT);
        info!("Plot {} : {:?}", self.label, self.rect);
        let mut child_ui = ui.child_ui(self.rect, layout);
        let _r  = pl.show(&mut child_ui, |plot_ui| {
            plot_ui.line(line);
        })
        .response;
        Ok(())
    }
```

**Comment by YgorSouza** [comment]
Normally the plot doesn't request a repaint by itself. It doesn't in the demo app nor in the examples. You can use [repaint_causes](https://docs.rs/egui/0.26.0/egui/struct.Context.html#method.repaint_causes) to find out why it is repainting.

**Comment by vortex314** [comment]
Just got a little issue there for finding the causes, I use also egui_gauge which is reliant on egui 0.25.0 which doesn't have repaint_causes function. This is only available since 0.26.0. Digging deeper....

**Comment by vortex314** [comment]
Strange thing happened, I build egui_gauge with 0.26.0 instead of 0.25.0 ,  also my project with egui 0.26.0 and suddenly the continuous repainting is gone as a side-effect of egui_plot.  Looks to be a version difference 0.25.0 to 0.26.0 that solved the issue.
